tls/s2n_server_new_session_ticket.c

```diff
@@ -74,3 +75,75 @@ int s2n_server_nst_send(struct s2n_connection *conn)
 
     return 0;
 }
+
+int s2n_tls13_server_nst_send(struct s2n_connection *conn)
+{
+    notnull_check(conn);
+
+    /* Write message type because session resumption in TLS13 is a post-handshake message */
+    GUARD(s2n_stuffer_write_uint8(&conn->handshake.io, TLS_SERVER_NEW_SESSION_TICKET));
+
+    struct s2n_stuffer_reservation message_size = { 0 };
+    GUARD(s2n_stuffer_reserve_uint24(&conn->handshake.io, &message_size));
+
+    /* Ticket lifetime is the minimum between key lifetime, session state lifetime, and one week */
+    uint32_t key_lifetime_in_secs = (conn->config->encrypt_decrypt_key_lifetime_in_nanos +
+            conn->config->decrypt_key_lifetime_in_nanos) / ONE_SEC_IN_NANOS;
```

Comment:
You should only need to consider decrypt_key_lifetime_in_nanos. To use a session, we only have to be able to decrypt it, not to encrypt it again.